ycmd/__main__.py

```diff
@@ -111,9 +111,10 @@ def SetupLogging( log_level ):
                        level = numeric_level )
 
 def SetupOptions( options_file ):
-  options = ( json.load( open( options_file, 'r' ) )
-              if options_file
-              else user_options_store.DefaultOptions() )
+  options = user_options_store.DefaultOptions()
+  if options_file:
+    user_options = json.load( open( options_file, 'r' ) )
+    options.update( user_options )
```

Comment:
@Valloric I added code for loading the default values first and then update the dictionary with user_options if given as server commandline argument.